lib/libc/common/source/time/clock.c

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2023 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <time.h>
+
+#include <zephyr/kernel.h>
+
+#ifndef CLOCKS_PER_SEC
+#error libc does not define CLOCKS_PER_SEC which is required by ISO
+#endif
+
+clock_t clock(void)
+{
+	return k_uptime_ticks();
```

Comment:
Oh, sorry, you either need to set `CLOCKS_PER_SEC` to the same as `CONFIG_SYS_CLOCK_TICKS_PER_SEC`, or you need to convert from `CONFIG_SYS_CLOCK_TICKS_PER_SEC` to `CLOCKS_PER_SEC`. POSIX says that `CLOCKS_PER_SEC` must be set to 1 million, but C doesn't have an opinion. If you're letting the C library set the value, it's likely to be 1 million.

Reply:
I'm thinking, when we eventually do have the correct bits in place for e.g. k_clock_gettime(), that both POSIX and libc could use a common code path that is in neither POSIX or libc.

Sort of again pushing the cyclic dependency to a mutual 3rd dependency like what will hopefully be done soon in net.

Reply:
In any case, did you want to fix this so we can get this API merged?